docs/reference/errors.md

```diff
@@ -0,0 +1,63 @@
+| PLUGIN_NOT_FOUND | plugin not found | No plugin was found. Verify the required plugin is supported by Ratify and check the plugin name is entered correctly. |
+| DOWNLOAD_PLUGIN_FAILURE | download plugin failure | Failed to download plugin. Please verify the provided plugin configuration is correct and check the error details for further investigation. Refer to https://github.com/deislabs/ratify/blob/main/docs/reference/dynamic-plugins.md for more information. |
+| CERT_INVALID | cert invalid | The certificate is invalid. Please verify the provided inline certificates or certificates fetched from key vault are in valid format. Refer to https://github.com/deislabs/ratify/blob/main/docs/reference/crds/certificate-stores.md for more information. |
+| PROVIDER_NOT_FOUND | provider not found | No provider was found. Please verify that the necessary policy provider has been registered and that the provider's name has been entered correctly. |
+
+#### Verifier errors
+| Code | Description | Trouble Shooting Guide |
+| ---- | ----------- | ---------------------- |
+| VERIFY_REFERENCE_FAILURE | verify reference failure | Generic error returned when the verifier fails to verify the reference. Please check the error details for more information. |
+| VERIFY_SIGNATURE_FAILURE | verify signature failure | Verifier failed to verify signature. Please check the error details from the verifier plugin and refer to plugin's documentation for more details. |
+| SIGNATURE_NOT_FOUND | signature not found | No signature was found. Please validate the verifying artifact has attached any expected signatures. |
+
+#### Certificate Store errors
+
+| Code | Description  | Trouble Shooting Guide |
+| ---- | -----------  | ---------------------- |
+| KEY_VAULT_OPERATION_FAILURE | Key vault operation failed. Please validate correct key vault configuration is provided or check the error details for further investigation. | 
+
+#### Referrer Store errors
+| Code | Description  | Trouble Shooting Guide |
+| ---- | -----------  | ---------------------- |
+| LIST_REFERRERS_FAILURE | list referrers failure | Generic error returned when the referrer store fails to list the referrers. Refer to https://github.com/deislabs/ratify/blob/main/docs/reference/store.md#listreferrers for more details. |
+| GET_SUBJECT_DESCRIPTOR_FAILURE | get subject descriptor failure | Generic error returned when the referrer store fails to get the subject descriptor. Refer to https://github.com/deislabs/ratify/blob/main/docs/reference/store.md#getsubjectdescriptor for more details. |
+| GET_REFERRER_MANIFEST_FAILURE | get reference manifest failure | Generic error returned when the referrer store fails to get the reference manifest. Refer to https://github.com/deislabs/ratify/blob/main/docs/reference/store.md#getreferencemanifest for more details. |
```

Comment:
I think we can remove the "generic error". Just leave it as "referrer store fails to get the reference manifest. Refer to https://github.com/deislabs/ratify/blob/main/docs/reference/store.md#getreferencemanifest for more details. "